### PR TITLE
Validation for monetized enabled org for enabling teams for ApigeeX

### LINKED ISSUE #936: Apigee X: Monetization vs Teams requirement check is a dead code (?)
## Description
https://github.com/apigee/apigee-edge-drupal/blob/8466a97f3cacf4224b473b3b8d10891476cbb5f6/modules/apigee_edge_teams/apigee_edge_teams.install#L44-L57

## Apigee Info
Apigee X with Monetization enabled.

## Steps to Reproduce
Enable the Apigee Edge Teams module on a Monetization enabled Apigee X org.

## Actual Behavior
The module was enabled with no requirement error.

## Expected Behavior
Well the Teams functionality is available on Apigee X as AppGroups but based on our discussion the AppGroup+Monetization integration missing at this moment, so probably the **fixed** requirement check should be added to https://github.com/apigee/apigee-m10n-drupal/blob/2.1.0/modules/apigee_m10n_teams/apigee_m10n_teams.install

## Version Info
3.0.1

### PR DESCRIPTION
**Pull request by shishir-intelli**
Closes https://github.com/apigee/apigee-edge-drupal/issues/936

### CODE REVIEW

File: modules/apigee_m10n_teams/apigee_m10n_teams.install [hunk 1]
```diff
@@ -30,3 +30,56 @@ function apigee_m10n_teams_install() {
...
+            t("The Apigee Monetization Teams module functionality is not available for monetization enabled org on Apigee X / Hybrid because <a href=':url' target='_blank'>AppGroup APIs are not supported in Apigee X / Hybrid orgs with monetization enabled</a>.", $url);
+          $requirements['apigee_edge_teams_not_supported'] = [
+            'title' => t('Apigee Edge Teams'),
+            'description' => $message,
+            'severity' => REQUIREMENT_ERROR,
+          ];
+        }
+      } else {
+        // Edge company APIs are not supported in Apigee Hybrid orgs.
+        if ($organization && !OrganizationFeatures::isCompaniesFeatureAvailable($organization)) {
```
Comment (giteshk) [inline modules/apigee_m10n_teams/apigee_m10n_teams.install]: Why are we checking companies ?

Comment (shishir-intelli) [reply modules/apigee_m10n_teams/apigee_m10n_teams.install]: For Edge, we already had this here https://github.com/apigee/apigee-edge-drupal/blob/fd665534475b3cf17b0520a01228e5ed490d8a1f/modules/apigee_edge_teams/apigee_edge_teams.install#L43, 
Want to confirm if Edge company APIs support Apigee Hybrid orgs (_Edge org with `features.hybrid.enabled` enabled_.) If this is no more needed then we can remove it.

Comment (giteshk) [reply modules/apigee_m10n_teams/apigee_m10n_teams.install]: **Apigee Edge** has support for Companies.. I don't think you can disable Companies API. 

Maybe the check is for **Monetization Teams** module ?

File: modules/apigee_m10n_teams/apigee_m10n_teams.install [hunk 2]
```diff
@@ -30,3 +30,56 @@ function apigee_m10n_teams_install() {
...
+          $url = [
+            ':url' => 'https://cloud.google.com/apigee/docs/api-platform/get-started/compare-apigee-products#unsupported-apis',
+          ];
+          $message = ($phase == 'runtime') ?
+            t("The Apigee Monetization Teams module functionality is not available for your org and should be uninstalled, because <a href=':url' target='_blank'>Edge company APIs are not supported for Apigee Hybrid orgs</a>.", $url) :
```
Comment (giteshk) [inline modules/apigee_m10n_teams/apigee_m10n_teams.install]: We should not callout "Apigee Hybrid orgs" in this section, if it is related to *Apigee Edge*

File: modules/apigee_m10n_teams/apigee_m10n_teams.install (hunk 1 above)
Comment (shishir-intelli) [reply modules/apigee_m10n_teams/apigee_m10n_teams.install]: No, it is checking for hybrid feature in the Edge Org, which don't support companies API right?

Comment (shishir-intelli) [reply modules/apigee_m10n_teams/apigee_m10n_teams.install]: If `features.hybrid.enabled` property is in ApigeeX Org and not in Edge Org then we don't need the else part.

**Review by divya-intelli — APPROVED** [review]
LGTM

**Pull request merged**